contracts/token/ERC865/ERC865Token.sol

```diff
@@ -0,0 +1,366 @@
+     * @param _spender address The address which will spend the funds.
+     * @param _subtractedValue uint256 The amount of tokens to decrease the allowance by.
+     * @param _fee uint256 The amount of tokens paid to msg.sender, by the owner.
+     * @param _nonce uint256 Presigned transaction number.
+     */
+    function decreaseApprovalPreSigned(
+        bytes _signature,
+        address _spender,
+        uint256 _subtractedValue,
+        uint256 _fee,
+        uint256 _nonce
+    )
+        public
+        returns (bool)
+    {
+        require(_spender != address(0));
+        require(signatures[_signature] == false);
+
+        bytes32 hashedTx = decreaseApprovalPreSignedHashing(address(this), _spender, _subtractedValue, _fee, _nonce);
+        address from = recover(hashedTx, _signature);
+        require(from != address(0));
+
+        uint oldValue = allowed[from][_spender];
+        if (_subtractedValue > oldValue) {
+            allowed[from][_spender] = 0;
+        } else {
+            allowed[from][_spender] = oldValue.sub(_subtractedValue);
+        }
+        balances[from] = balances[from].sub(_fee);
+        balances[msg.sender] = balances[msg.sender].add(_fee);
+        signatures[_signature] = true;
+
+        Approval(from, _spender, _subtractedValue);
+        Transfer(from, msg.sender, _fee);
+        ApprovalPreSigned(from, _spender, msg.sender, allowed[from][_spender], _fee);
+        return true;
+    }
+
+    /**
+     * @notice Transfer tokens from one address to another
+     * @param _signature bytes The signature, issued by the spender.
+     * @param _from address The address which you want to send tokens from.
+     * @param _to address The address which you want to transfer to.
+     * @param _value uint256 The amount of tokens to be transferred.
+     * @param _fee uint256 The amount of tokens paid to msg.sender, by the spender.
+     * @param _nonce uint256 Presigned transaction number.
+     */
+    function transferFromPreSigned(
+        bytes _signature,
+        address _from,
+        address _to,
+        uint256 _value,
+        uint256 _fee,
+        uint256 _nonce
+    )
+        public
+        returns (bool)
+    {
+        require(_to != address(0));
+        require(signatures[_signature] == false);
+
+        bytes32 hashedTx = transferFromPreSignedHashing(address(this), _from, _to, _value, _fee, _nonce);
+
+        address spender = recover(hashedTx, _signature);
+        require(spender != address(0));
+
+        balances[_from] = balances[_from].sub(_value);
+        balances[_to] = balances[_to].add(_value);
+        allowed[_from][spender] = allowed[_from][spender].sub(_value);
+
+        balances[spender] = balances[spender].sub(_fee);
+        balances[msg.sender] = balances[msg.sender].add(_fee);
+        signatures[_signature] = true;
+
+        Transfer(_from, _to, _value);
+        Transfer(spender, msg.sender, _fee);
```

Comment:
Does not it require to emit `TransferFromPreSigned` or `TransferPreSigned` event?